### PR TITLE
Adding AuthorRepository.cs and IAuthorRepository.cs

### PR DESCRIPTION
**Pull request by Tipskind**
New interface IAuthorRepository

New repository AuthorRepository

New Dependency injection:
builder.Services.AddScoped<AuthorRepository>();

Unit test adapted

Dublicated method FindAuthorByName in AuthorRepository and CheepRepository.

### CODE REVIEW

**Review by Nikolaj787 — CHANGES_REQUESTED** [review]
Overall really great changes to seperate the CheepRepository Author methods into a seperate repository. But there are some duplicate methods in both repositories. Maybe reference the author repository, so that we avoid redundancy? Then they will be completely seperate :D

**Review by niko391a — COMMENTED** [review]
Well thought out to refactor our repository into two, makes it easier to have a general overview.

**Review by Nikolaj787 — DISMISSED** [review]
Much better that the author methods now are seperated into a seperate repository!

File: src/Chirp.Infrastructure/Services/CheepService.cs [hunk 1]
```diff
@@ -52,12 +54,12 @@ public async Task<int> GetTotalPageNumber(string authorName = "")
...
     public async Task CreateAuthor(string authorName, string authorEmail)
     {
-        await _cheepRepository.CreateAuthor(authorName, authorEmail);
+        await _authorRepository.CreateAuthor(authorName, authorEmail);
     }
 
     public async Task<AuthorDTO>? FindAuthorByName(String name)
     {
-        return _cheepRepository.FindAuthorByNameDTO(name);
+        return _authorRepository.FindAuthorByNameDTO(name);
```
Comment (RasmusAChr) [inline src/Chirp.Infrastructure/Services/CheepService.cs:62]: Use await when invoking an asynchronous method call: `return await _authorRepository.FindAuthorByNameDTO(name);`

**Review by RasmusAChr — CHANGES_REQUESTED** [review]
Great extraction of author methods into their own repository. I like the consideration you have shown towards using the `AuthorRepository` in the `CheepRepository` to avoid redundancy. 

I have some minor changes to be requested.

The following methods in `AuthorRepository` are missing in `IAuthorRepository`:
- FindAuthorByNameDTO
- FindAuthorByName

In `CheepService` for the `FindAuthorByName`, remember to call it asynchronously by using `return await _authorRepository.FindAuthorByNameDTO(name);`**

File: src/Chirp.Core/Interfaces/IAuthorRepository.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using Chirp.Core.DTOs;
+
+namespace Chirp.Core.Interfaces;
+
+public interface IAuthorRepository
+{
+    Task CreateAuthor(string authorName, string authorEmail);
+    Task DeleteUser(AuthorDTO Author);
+    Task FollowAuthor(string userAuthor, string followedAuthor);
```
Comment (DragonFlyersx) [inline src/Chirp.Core/Interfaces/IAuthorRepository.cs:9]: Can we keep the camelcase also in the attributes of the methods

File: src/Chirp.Infrastructure/Repositories/AuthorRepository.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        public Author? FindAuthorByName(String name)
+        {
+            var author = (from a in _dbContext.Authors
+                where a.Name == name
+                select a).FirstOrDefault();
+            return author;
+        }
+        
+        // Find a user by their email
+        public Author? FindAuthorByEmail(string email)
```
Comment (DragonFlyersx) [inline src/Chirp.Infrastructure/Repositories/AuthorRepository.cs:43]: I believe this method is not used

Comment (DragonFlyersx) [reply src/Chirp.Infrastructure/Repositories/AuthorRepository.cs:43]: Should also return an authorDTO instead of author object

**Review by DragonFlyersx — CHANGES_REQUESTED** [review]
Otherwise nice job creating a new reporsitory for the author methods

**Review by niko391a — APPROVED** [review]

File: src/Chirp.Core/Interfaces/IAuthorRepository.cs (hunk 2 above)
Comment (RasmusAChr) [reply src/Chirp.Core/Interfaces/IAuthorRepository.cs:11]: Marcus have done it the correct way with writing [camel case](https://da.wikipedia.org/wiki/CamelCase). I think you might confuse it with [Pascal Case](https://www.theserverside.com/definition/Pascal-case).
- thisIsCamelCase
- ThisIsPascalCase

Comment (RasmusAChr) [reply src/Chirp.Core/Interfaces/IAuthorRepository.cs:11]: On that note, the methods should per standard be Camel Case aswell, but that's minor.

**Review by RasmusAChr — APPROVED** [review]
Good job implementing the fixes

**Review by DragonFlyersx — APPROVED** [review]
Nicely done

**Review by Nikolaj787 — APPROVED** [review]

**Review by niko391a — APPROVED** [review]

**Pull request merged**